### PR TITLE
Add SQL commands for local database operations

### PR DESCRIPTION
**Pull request by findolor**
> [!CAUTION]
> **Chained PR**
> Do not merge before https://github.com/rainlanguage/rain.orderbook/pull/2104

> [!NOTE]
> This PR https://github.com/rainlanguage/rainix/pull/88 also needs to be merged to enable sqlite3 command in nix shell

## Motivation

Previously, we could fetch and decode events but lacked the ability to convert decoded events to SQL
  format and create complete database dumps.

## Solution

  This PR adds two new CLI commands for local database SQL operations:

  1. decoded-events-to-sql - Converts decoded JSON events to SQL INSERT statements with configurable output file
  and end block parameters
  2. dump - Creates complete database dumps by setting up schema, importing data, and generating compressed SQL
  dumps

  Key changes include:
  - Added DecodedEventsToSql command with comprehensive test coverage for JSON parsing, file I/O, and error
  handling
  - Added DbDump command that orchestrates schema creation, data import, and database dumping with gzip compression
  - Enhanced .gitignore to exclude generated SQL files and database artifacts
  - Added sync status table to the database schema for tracking synchronization state
  - Both commands include extensive unit tests covering success scenarios, error conditions, and edge cases

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [ ] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

## Summary by CodeRabbit

- New Features
  - Added CLI commands to convert decoded event JSON into SQL and to export the local DB as a gzipped SQL dump.
  - Added a sync_status table to track last synced block and timestamp.
  - Added deserialization support for decoded event types to enable round-trip JSON handling.

- Improvements
  - Improved default output path resolution and status messages for local DB/event commands.

- Chores
  - Expanded .gitignore to exclude generated local DB artifacts and SQL event outputs.

### CODE REVIEW

**Comment by findolor** [comment]
@coderabbitai review

File: crates/cli/src/commands/local_db/dump.rs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            .arg(".dump")
+            .output()?;
+
+        fs::write(&dump_file_path, output.stdout)?;
+
+        Command::new("gzip")
+            .arg("-f")
+            .arg(&dump_file_path)
+            .status()?;
+
```
[the remark this answers is not in the document]
Comment (findolor) [reply crates/cli/src/commands/local_db/dump.rs:55]: this is fine we are going to get these errors any way

**Review by 0xgleb — APPROVED** [review]
